app/scenes/Settings/components/ConnectedButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import { s } from "@shared/styles";
+import Button, { Props as ButtonProps } from "~/components/Button";
+import ConfirmationDialog from "~/components/ConfirmationDialog";
+import Text from "~/components/Text";
+import useStores from "~/hooks/useStores";
+
+type Props = ButtonProps<HTMLButtonElement> & {
+  confirmationMessage: React.ReactNode;
+  onClick: () => Promise<void> | void;
+};
+
+export function ConnectedButton({
```

Comment:
<img width="166" alt="image" src="https://github.com/outline/outline/assets/380914/9fd7ae62-5992-4271-b767-0177c65c31cf">

Connected button with confirmation dialog to disconnect